app/models/applicant.rb

```diff
@@ -91,6 +91,10 @@ def delegate_field_to item, field_name
 
   def value_for_field field_name
     case field_name
+    when /^(.*)_or_(.*)$/
+      value = value_for_field($1)
+      return $2.capitalize.sub('_', ' ') if value == nil || value == ''
```

Comment:
Try using "if value.blank?" here.

Also, it would be nice to see a unit test for this in test/unit/field_filling_test.rb.

Note that #value_for_field can also return an instance of UnknownField (defined in lib/unknown_field.rb), which indicates that the name wasn't something that we support.  You should test what happens in this case, too.  Should calling #field with "asdfghjkl_or_foo" return the string "foo" or an empty string?